server/src/main/java/io/druid/server/http/CoordinatorDynamicConfigsResource.java

```diff
@@ -76,15 +76,20 @@ public Response getDynamicConfigs()
   // default value is used for backwards compatibility
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
-  public Response setDynamicConfigs(final CoordinatorDynamicConfig dynamicConfig,
+  public Response setDynamicConfigs(final CoordinatorDynamicConfig.Builder dynamicConfigBuilder,
                                     @HeaderParam(AuditManager.X_DRUID_AUTHOR) @DefaultValue("") final String author,
                                     @HeaderParam(AuditManager.X_DRUID_COMMENT) @DefaultValue("") final String comment,
                                     @Context HttpServletRequest req
   )
   {
+    CoordinatorDynamicConfig current = manager.watch(
+        CoordinatorDynamicConfig.CONFIG_KEY,
+        CoordinatorDynamicConfig.class
+    ).get();
+
     if (!manager.set(
         CoordinatorDynamicConfig.CONFIG_KEY,
-        dynamicConfig,
+        current == null ? dynamicConfigBuilder.build() : dynamicConfigBuilder.build(current),
```

Comment:
ordering is correct. "current" is what is stored in DB now ... so if there is nothing in db (current == null case) then use generic defaults or use values in "current" as defaults.